bin/picca_coadd_zint.py

```diff
@@ -0,0 +1,458 @@
+        healpixs = hdul[2]['HEALPID'][:]
+
+        # get values of header entries to check with other files
+        header = hdul[1].read_header()
+        headers_to_check_match_values = {h:header[h] for h in headers_to_check_match}
+
+    # initialise header quantities
+    r_par_min = 1.e6
+    r_par_max = -1.e6
+    r_trans_max = -1.e6
+    z_cut_min = 1.e6
+    z_cut_max = -1.e6
+
+    # check to see if all files have the same HEALPix pixels
+    same_healpixs = True
+    for file in fi:
+        with fitsio.FITS(file) as hdul:
+            if len(hdul[2]['HEALPID'][:]) != len(healpixs):
+                same_healpixs *= False
+            elif (hdul[2]['HEALPID'][:] != healpixs).all():
+                same_healpixs *= False
+
+    # if so, initialise our xi and weights as arrays
+    if same_healpixs:
+        with fitsio.FITS(file) as hdul:
+            xi = hdul[2]['DA'][:] * 0
+            weights = hdul[2]['WE'][:] * 0
+    # otherwise, use dictionaries to match HEALPix pixels carefully (but slower)
+    else:
+        xi = {}
+        weights = {}
+
+    # loop over files
+    for file in fi:
+
+        # open the file and read the header
+        userprint("coadding file {}".format(file))
+        hdul = fitsio.FITS(file)
+        header = hdul[1].read_header()
+
+        # check that the header properties match those from the first file
+        for h in headers_to_check_match:
+            assert header[h] == headers_to_check_match_values[h]
+
+        # add weighted contributions from this file to coadd variables
+        weights_aux = hdul[2]['WE'][:]
+        weights_total_aux = weights_aux.sum(axis=0)
+        r_par += hdul[1]['RP'][:] * weights_total_aux
+        r_trans += hdul[1]['RT'][:] * weights_total_aux
+        z += hdul[1]['Z'][:] * weights_total_aux
+        num_pairs += hdul[1]['NB'][:]
+        weights_total += weights_total_aux
+
+        # update values to go in coadd header
+        r_par_min = np.min([r_par_min,header['RPMIN']])
+        r_par_max = np.max([r_par_max,header['RPMAX']])
+        r_trans_max = np.max([r_trans_max,header['RTMAX']])
+        z_cut_min = np.min([z_cut_min,header['ZCUTMIN']])
+        z_cut_max = np.max([z_cut_max,header['ZCUTMAX']])
+
+        # add xi and weights information to initialised structures
+        if same_healpixs:
+            xi += hdul[2]["DA"][:] * weights_aux
+            weights += weights_aux
+        else:
+            healpixs = hdul[2]['HEALPID'][:]
+            for index, healpix in enumerate(healpixs):
+                userprint(" -> coadding healpix {}".format(healpix),end='\r')
+                if healpix in xi:
+                    xi[healpix] += hdul[2]["DA"][:][index] * weights_aux[index]
+                    weights[healpix] += weights_aux[index, :]
+                else:
+                    xi[healpix] = hdul[2]["DA"][:][index] * weights_aux[index]
+                    weights[healpix] = weights_aux[index]
+            print('')
```

Comment:
I think we should remove this print or, at least, convert it to `userprint`

Reply:
I think this print is needed otherwise subsequent printed output is placed on top of the information about which pixels are being coadded. Will change to `userprint` in a commit thought